idaes/models_extra/gas_solid_contactors/unit_models/moving_bed.py

```diff
@@ -433,12 +498,49 @@ def build(self):
         # Get units meta data from property packages
         units_meta_solid = solid_phase.property_package.get_metadata().get_derived_units
 
-        # Create a unit model length domain
-        self.length_domain = ContinuousSet(
-            bounds=(0.0, 1.0),
-            initialize=self.config.length_domain_set,
-            doc="Normalized length domain",
-        )
+        # In this branch, we assume that either both gas and solid
+        # transformation schemes are specified, or neither are specified.
+        if (
+            # solid and gas transformation schemes are specified
+            (self.config.solid_transformation_scheme is not None)
+            and (self.config.gas_transformation_scheme is not None)
```

Comment:
A minor question, but would it be better to just check that the transformation schemes are different here - i.e. the real point is that if they are different then we need separate domains.

However, going a step further, I would suggest we consider just having two domains at all times (regardless of the transformations used). In the current approach, the model structure is different depending on the choices made, which may have unforeseen consequences, so it might be worth just using two domains at all times to avoid this.

Reply:
My thought was that, if schemes are provided for solid and gas, they should have different length domains. In this way, whether or not we have two length domains depends on whether these arguments were provided, rather than the value of the arguments. I don't feel strongly about this, however.

I would rather not *always* have two length domains, however, as this would break code that relies on the phases having the same length domain (I have written such code myself). I agree that always having two length domains is a cleaner solution, however.